### DISCUSSION TITLE
Setting "terminal.integrated.defaultProfile.linux": "zsh" does not work on relaunch, bash shell is spawned instead

### DISCUSSION THREAD

**Discussion by kompasbank-tobias** [community/community#11868] [Codespaces]
Setting "terminal.integrated.defaultProfile.linux": "zsh" in settings.json works on new terminals created in Codespaces but the initial terminal always seems to be bash (initial terminal = terminal spawned after wakeup of a Codespace).

Anyone else experience this annoying behaviour?

Regards,
Tobias

**Comment by nilsml** [comment]
Experiencing the same thing. Have not found a way to fix it yet. Pretty annoying!

**Comment by AndriiTsok** [comment]
We experiencing the same issue, zsh is set up in the devcontainer.json file as vscode setting customizations. But it opens bash on start.

**Comment by huw** [comment]
Tried today, you can fix this by changing the default shell on in your Dockerfile. This will differ from user-to-user, but for me:

```dockerfile
RUN apt-get update && apt-get install -y zsh

RUN groupadd --gid 1000 node \
  && useradd --uid 1000 --gid node --shell /bin/zsh --create-home node
```

**Comment by JasonToups** [comment]
I'm having the same issue, but I'm running this on a Steam Deck, so I'm using Arch Linux SteamOS 3.0.
